test/e2e/kubevirt.go

```diff
@@ -1431,12 +1431,15 @@ runcmd:
 				By("setting up the localnet underlay")
 				nodes := ovsPods(clientSet)
 				Expect(nodes).NotTo(BeEmpty())
-				defer func() {
-					By("tearing down the localnet underlay")
-					Expect(teardownUnderlay(nodes)).To(Succeed())
-				}()
+				DeferCleanup(func() {
+					if e2eframework.TestContext.DeleteNamespace && (e2eframework.TestContext.DeleteNamespaceOnFailure || !CurrentSpecReport().Failed()) {
+						By("tearing down the localnet underlay")
+						Expect(teardownUnderlay(nodes)).To(Succeed())
+					}
+				})
 
 				const secondaryInterfaceName = "eth1"
+				teardownUnderlay(nodes)
```

Comment:
I don't understand this call in this position. You're tearing down the underlay before the test even begins. 

And you're not handling the error.

Reply:
Sometimes test do not reach defer, so I force delete it to be sure that it do not exist before creating it.

Reply:
ah. and we setup the underlay again afterwards ?...

Reply:
Yes

Reply:
so I think that:
- you should handle the error. 
- you shouldn't do this here. Use an AfterEach block if you really need. 
- Isn't this redundant w/ the `DeferCleanup` you're adding ? 

At the end of the day, if the tests don't reach the defer you have a problem in the test - IMHO you should the failing test, not correct the subsequent ones.